### PR TITLE
[CI Visibility] Add .NET manual api documentation

### PR DESCRIPTION
**Pull request by tonyredondo**
### What does this PR do?

This PR adds the CI Visibility Manual Api documentation for .NET

### Motivation

The feature was released but the documentation is missing.

### Preview 
https://docs-staging.datadoghq.com/tony/add-dotnet-manual-api/continuous_integration/tests/dotnet

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/tests/dotnet.md [hunk 1]
```diff
@@ -286,6 +285,575 @@ To use the custom instrumentation in your .NET application:
...
+
+### Test module
+
+A test module represents the .NET assembly that includes the tests.
+
+To start a test module, call `TestModule.Create()` and pass the name of the module or .NET assembly name where the tests are located.
+
+When all your tests have finished, call `module.Close()` or `module.CloseAsync()`, which forces the library to send all remaining test results to the backend.
+
+### Test Suites
```
Comment (albertvaka) [inline content/en/continuous_integration/tests/dotnet.md:306]:
```suggestion
### Test suites
```

**Review by albertvaka — APPROVED** [review]
Reworded some sentences but overall LGTM.

**Comment by hestonhoffman** [comment]
👋 I've added a [docs card](https://datadoghq.atlassian.net/browse/DOCS-5279) for this one

File: content/en/continuous_integration/tests/dotnet.md [hunk 2]
```diff
@@ -286,6 +285,575 @@ To use the custom instrumentation in your .NET application:
...
 
+## Manual testing API
+
+<div class="alert alert-warning">
+  <strong>Note:</strong> To use the manual testing API, you must add the <code>Datadog.Trace</code> NuGet package in the target .NET project.
+</div>
+
+If you use XUnit, NUnit or MSTest with your .NET projects, CI Visibility will automatically instruments them and sends the results to the Datadog backend. If you use an unsupported testing framework or if you have a different testing mechanism, you can instead use the CI Visibility manual testing API, which also reports test results to the backend.
+
+The API is based around three concepts: *test module*, *test suites*, and *tests*.
```
Comment (drichards-87) [inline content/en/continuous_integration/tests/dotnet.md:296]:
```suggestion
The API is based around three concepts: test module, test suites, and tests.
```

**Review by drichards-87 — CHANGES_REQUESTED** [review]

**Review by drichards-87 — APPROVED** [review]
@tonyredondo PR is approved. Should I merge it now?

**Comment by tonyredondo** [comment]
> @tonyredondo PR is approved. Should I merge it now?

Yes please! Thank you!

**Pull request merged**